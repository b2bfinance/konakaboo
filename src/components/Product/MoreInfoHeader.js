import React from 'react';
import styled from 'styled-components';
import Hidden from '@material-ui/core/Hidden';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';

export const Wrapper = styled.div`
  padding: 16px;
  border-bottom: 2px solid ${props => props.theme.productBorder};
`;

export const LogoWrapper = styled.div`
  border: 2px solid ${props => props.theme.productBorder};
  display: flex;
  height: 100px;
  width: 150px;
`;

export const Logo = styled.img`
  max-width: 100%;
  max-height: 100%;
  margin: auto;
`;

export const CloseButton = styled.div`
  position: absolute;
  top: 8px;
  right: 8px;
`;

export default ({ title, brand, logo, onClose }) => (
  <Wrapper>
    <Grid container spacing={4}>
      <Hidden xsDown>
        <Grid item sm={3} md={3} xl={2}>
          <LogoWrapper>
            <Logo src={logo} alt={brand} />
          </LogoWrapper>
        </Grid>
      </Hidden>
      <Grid item sm={9} md={9} xl={10} mt={1}>
        <Grid container>
          <Grid item xs={12}>
            <Typography variant="h4">{brand}</Typography>
          </Grid>
          <Grid item xs={12}>
            <Typography variant="subtitle1">{title}</Typography>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
    <CloseButton>
      <IconButton aria-label="Close" onClick={onClose}>
        <CloseIcon />
      </IconButton>
    </CloseButton>
  </Wrapper>
);
