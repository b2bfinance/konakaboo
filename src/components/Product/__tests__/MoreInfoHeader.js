import React from 'react';
import { render, mount } from 'enzyme';
import MoreInfoHeader, {
  Wrapper,
  LogoWrapper,
  Logo,
  CloseButton
} from '../MoreInfoHeader';
import IconButton from '@material-ui/core/IconButton';

describe('Wrapper', () => {
  test('renders correctly', () => {
    expect(render(<Wrapper />)).toMatchSnapshot();
  });
});

describe('LogoWrapper', () => {
  test('renders correctly', () => {
    expect(render(<LogoWrapper />)).toMatchSnapshot();
  });
});

describe('Logo', () => {
  test('renders correctly', () => {
    expect(render(<Logo />)).toMatchSnapshot();
  });
});

describe('CloseButton', () => {
  test('renders correctly', () => {
    expect(render(<CloseButton />)).toMatchSnapshot();
  });
});

describe('MoreInfoHeader', () => {
  let Component;
  let wrapper;
  let onClose;

  beforeAll(() => {
    const product = stubData.products.item;

    onClose = jest.fn();

    Component = (
      <MoreInfoHeader
        onClose={onClose}
        title={product.title}
        brand={product.brand}
        logo={product.links.logo}
      />
    );

    wrapper = mount(Component);
  });

  afterAll(() => {
    wrapper.unmount();
  });

  test('renders correctly', () => {
    expect(render(Component)).toMatchSnapshot();
  });

  test('should call onClose when close triggered', () => {
    wrapper.find(IconButton).simulate('click');
    expect(onClose).toHaveBeenCalledTimes(1);
  });
});
