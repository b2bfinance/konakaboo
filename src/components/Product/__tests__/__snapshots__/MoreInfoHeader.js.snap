// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton renders correctly 1`] = `
.c0 {
  position: absolute;
  top: 8px;
  right: 8px;
}

<div
  class="c0"
/>
`;

exports[`Logo renders correctly 1`] = `
.c0 {
  max-width: 100%;
  max-height: 100%;
  margin: auto;
}

<img
  class="c0"
/>
`;

exports[`LogoWrapper renders correctly 1`] = `
.c0 {
  border: 2px solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100px;
  width: 150px;
}

<div
  class="c0"
/>
`;

exports[`MoreInfoHeader renders correctly 1`] = `
.c0 {
  padding: 16px;
  border-bottom: 2px solid;
}

.c1 {
  position: absolute;
  top: 8px;
  right: 8px;
}

<div
  class="c0"
>
  <div
    class="MuiGrid-container-1 MuiGrid-spacing-xs-32-26"
  >
    <div
      class="MuiGrid-item-2"
    >
      <div
        class="MuiGrid-container-1"
      >
        <div
          class="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <h4
            class="MuiTypography-root-98 MuiTypography-h4-113"
          >
            Landbay
          </h4>
        </div>
        <div
          class="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <h6
            class="MuiTypography-root-98 MuiTypography-subtitle1-116"
          >
            Landbay - Buy To Let Wizards
          </h6>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c1"
  >
    <button
      aria-label="Close"
      class="MuiButtonBase-root-140 MuiIconButton-root-134"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label-139"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-143"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`Wrapper renders correctly 1`] = `
.c0 {
  padding: 16px;
  border-bottom: 2px solid;
}

<div
  class="c0"
/>
`;
